src/Microsoft.Health.Dicom.SqlServer/Features/Query/SqlQueryStoreV27.cs

```diff
@@ -0,0 +1,142 @@
+        SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+        ILogger<ISqlQueryStore> logger)
+        : base(sqlConnectionWrapperFactory, logger)
+    {
+    }
+
+    public override async Task<IReadOnlyCollection<StudyResult>> GetStudyResultAsync(
+        int partitionKey,
+        IReadOnlyCollection<long> versions,
+        CancellationToken cancellationToken)
+    {
+        var results = new List<StudyResult>();
+
+        using (SqlConnectionWrapper sqlConnectionWrapper = await SqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken))
+        using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand())
+        {
+            List<WatermarkTableTypeRow> versionRows = versions.Select(i => new WatermarkTableTypeRow(i)).ToList();
+            VLatest.GetStudyResult.PopulateCommand(
+                sqlCommandWrapper,
+                partitionKey,
+                versionRows);
+
+            using (var reader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+            {
+                while (await reader.ReadAsync(cancellationToken))
+                {
+                    (string rStudyInstanceUid,
+                    string rPatientId,
+                    string rPatientName,
+                    string rReferringPhysicianName,
+                    DateTime? rStudyDate,
+                    string rStudyDescription,
+                    string rAccessionNumber,
+                    DateTime? rPatientBirthDate,
+                    string rModalitiesInStudy,
+                    int rNumberofStudyRelatedInstances
+                     ) = reader.ReadRow(
+                       VLatest.Study.StudyInstanceUid,
+                       VLatest.Study.PatientId,
+                       VLatest.Study.PatientName,
+                       VLatest.Study.ReferringPhysicianName,
+                       VLatest.Study.StudyDate,
+                       VLatest.Study.StudyDescription,
+                       VLatest.Study.AccessionNumber,
+                       VLatest.Study.PatientBirthDate,
+                       new NVarCharColumn("ModalitiesInStudy", 4000),
+                       new IntColumn("NumberofStudyRelatedInstances"));
+
+                    results.Add(new StudyResult()
+                    {
+                        StudyInstanceUid = rStudyInstanceUid,
+                        PatientId = rPatientId,
+                        PatientName = rPatientName,
+                        ReferringPhysicianName = rReferringPhysicianName,
+                        StudyDate = rStudyDate,
+                        StudyDescription = rStudyDescription,
+                        AccessionNumber = rAccessionNumber,
+                        PatientBirthDate = rPatientBirthDate,
+                        ModalitiesInStudy = rModalitiesInStudy.Split(',', StringSplitOptions.RemoveEmptyEntries).Distinct().ToArray(),
```

Comment:
In a corner case if the returned value is null, this will result in null value exception if they do includeField=ModalityInStudy and use different filter conditions.